src/main/java/fi/metatavu/kuntaapi/server/integrations/AbstractIdProvider.java

```diff
@@ -0,0 +1,300 @@
+package fi.metatavu.kuntaapi.server.integrations;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.inject.Inject;
+
+import fi.metatavu.kuntaapi.server.controllers.IdentifierController;
+import fi.metatavu.kuntaapi.server.id.AnnouncementId;
+import fi.metatavu.kuntaapi.server.id.AttachmentId;
+import fi.metatavu.kuntaapi.server.id.BannerId;
+import fi.metatavu.kuntaapi.server.id.BaseId;
+import fi.metatavu.kuntaapi.server.id.ContactId;
+import fi.metatavu.kuntaapi.server.id.ElectronicServiceChannelId;
+import fi.metatavu.kuntaapi.server.id.EmergencyId;
+import fi.metatavu.kuntaapi.server.id.EventId;
+import fi.metatavu.kuntaapi.server.id.FileId;
+import fi.metatavu.kuntaapi.server.id.FragmentId;
+import fi.metatavu.kuntaapi.server.id.IdProvider;
+import fi.metatavu.kuntaapi.server.id.IdType;
+import fi.metatavu.kuntaapi.server.id.IncidentId;
+import fi.metatavu.kuntaapi.server.id.JobId;
+import fi.metatavu.kuntaapi.server.id.MenuId;
+import fi.metatavu.kuntaapi.server.id.MenuItemId;
+import fi.metatavu.kuntaapi.server.id.NewsArticleId;
+import fi.metatavu.kuntaapi.server.id.OrganizationBaseId;
+import fi.metatavu.kuntaapi.server.id.OrganizationId;
+import fi.metatavu.kuntaapi.server.id.PageId;
+import fi.metatavu.kuntaapi.server.id.PhoneServiceChannelId;
+import fi.metatavu.kuntaapi.server.id.PrintableFormServiceChannelId;
+import fi.metatavu.kuntaapi.server.id.PublicTransportAgencyId;
+import fi.metatavu.kuntaapi.server.id.PublicTransportRouteId;
+import fi.metatavu.kuntaapi.server.id.PublicTransportScheduleId;
+import fi.metatavu.kuntaapi.server.id.PublicTransportStopId;
+import fi.metatavu.kuntaapi.server.id.PublicTransportStopTimeId;
+import fi.metatavu.kuntaapi.server.id.PublicTransportTripId;
+import fi.metatavu.kuntaapi.server.id.ServiceId;
+import fi.metatavu.kuntaapi.server.id.ServiceLocationServiceChannelId;
+import fi.metatavu.kuntaapi.server.id.ShortlinkId;
+import fi.metatavu.kuntaapi.server.id.TileId;
+import fi.metatavu.kuntaapi.server.id.WebPageServiceChannelId;
+import fi.metatavu.kuntaapi.server.persistence.model.Identifier;
+
+@SuppressWarnings ("squid:S3306")
+public abstract class AbstractIdProvider implements IdProvider {
+
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private IdentifierController identifierController;
+  
+  public abstract String getSource();
+  public abstract boolean isSupportedType(IdType type);
+  
+  @Override
+  public boolean canTranslate(String source, String target) {
+    if (getSource().equals(source) && KuntaApiConsts.IDENTIFIER_NAME.equals(target)) {
+      return true;
+    }
+    
+    if (getSource().equals(target) && KuntaApiConsts.IDENTIFIER_NAME.equals(source)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)